### PR TITLE
MDAS-448 Metrics

### PR DESCRIPTION
**Pull request by mirrodi**
### Reference
**Documentation**: https://confluence.muenchen.de/pages/viewpage.action?pageId=259886340#id-5IntegrationsarchitekturundSchnittstellen(zuNachbarsystemen)-5.1.3.3Monitoring

Issues: https://jira.muenchen.de/browse/MDAS-448

### Definition of Done
*Advice: Unchecked checkboxes are not accepted by the build pipeline. Cross out any non-relevant item by using '~~'.*

- [x] Acceptance criteria are met
- [x] Build/Test workflow has successfully finished
- [x] Release notes are complemented
- [x] Documentation is complemented (operator manual, system specification)
~Frontend is locally smoke-tested~
- [x] Board is updated
~Infrastructure is adjusted~

Please delete all side branches after merging.

### CODE REVIEW

File: src/main/java/de/muenchen/mobidam/config/SecurityConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+
+/**
+ * The central class for configuration of all security aspects.
+ */
+@Configuration
+@Profile("!no-security")
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)
+@RequiredArgsConstructor
```
Comment (sfi2022) [inline src/main/java/de/muenchen/mobidam/config/SecurityConfiguration.java:44]: What is the constructor used for ?

File: src/main/java/de/muenchen/mobidam/config/MetricsConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    private final CamelContext camelContext;
+    private final FileSizeProcessor fileSizeProcessor;
+
+    private final Counter beginnCounter;
+    private final Counter endeCounter;
+    private final Counter fehlerCounter;
+    private final Counter erfolgCounter;
+    private final Counter warnungenCounter;
+    private final Gauge inflightExchanges;
+    private final Gauge maxFileSize;
```
Comment (sfi2022) [inline src/main/java/de/muenchen/mobidam/config/MetricsConfiguration.java:51]: Is there a reason for the mixture of German and English attribute names ?

File: src/main/java/de/muenchen/mobidam/config/SecurityConfiguration.java (hunk 1 above)
Comment (sfi2022) [reply src/main/java/de/muenchen/mobidam/config/SecurityConfiguration.java:44]: That ist not your fault. It was me. I removed the @RequiredArgsConstructor and the security still works. So i think you can remove it.

File: src/main/java/de/muenchen/mobidam/security/FileSizeProcessor.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+@Getter
+public class FileSizeProcessor implements Processor {
+
+    private long maxStreamSize = 0L;
+
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        InputStreamCache stream = exchange.getMessage().getBody(InputStreamCache.class);
+        stream.reset();
+        maxStreamSize = Math.max(maxStreamSize, stream.length());
```
Comment (sfi2022) [inline src/main/java/de/muenchen/mobidam/security/FileSizeProcessor.java:45]: This gives the correct value as long as we read an uncompressed file ?

Comment (mirrodi) [reply src/main/java/de/muenchen/mobidam/security/FileSizeProcessor.java:45]: Ja, aber die Art der Datei wird mit der Malware-Prüfung abgefrühstückt. Da werden nur bestimmte MIME Typen angenommen

**Pull request merged**